### ISSUE TITLE
Mongo explain results incorrectly returning an object instead of an array of objects

### ISSUE THREAD

**Issue by cbaker6** [parse-community/parse-server#7442]
### New Issue Checklist

- [ ] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [ ] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [ ] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [ ] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Failing Test Case / Pull Request

- [ ] 🤩 I submitted a PR with a fix and a test case.
- [ ] 🧐 I submitted a PR with a failing test case.

###  Environment

Server
- Parse Server version: `FILL_THIS_OUT`
- Operating system: `FILL_THIS_OUT`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `FILL_THIS_OUT`

Database
- System (MongoDB or Postgres): `FILL_THIS_OUT`
- Database version: `FILL_THIS_OUT`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `FILL_THIS_OUT`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `FILL_THIS_OUT`
- SDK version: `FILL_THIS_OUT`

**Comment by mtrezza** [comment]
Thanks for opening this issue. Is this an enhancement or a bug?

**Comment by cbaker6** [comment]
I would label this as a bug. The reason for this is in JS because the language isn’t strongly typed, but from an implementation standpoint, explain should have always returned an array since every other query returns an array when using find. The bug was discovered because of its use in the Swift SDK which is strongly typed

**Comment by mtrezza** [comment]
Isn't the ObjC SDK also strongly typed? How is that solved there?

**Comment by cbaker6** [comment]
> Isn't the ObjC SDK also strongly typed? How is that solved there?

I can check, but from memory, I don’t think the Objc SDK has the ability to explain

**Comment by mtrezza** [comment]
I think you are right, I just looked through it.

**Comment by mtrezza** [comment]
> This will break SDKs such as the JS SDK.

Do you mean that a change will inevitable include a breaking change for the JS SDK? Or is there anything broken currently in the JS SDK?

> In addition, this brings up another issue in the JS SDK which doesn't support explain for first queries.

Is this related to `results` not being an array, or a completely separate issue?

**Comment by cbaker6** [comment]
> Do you mean that a change will inevitable include a breaking change for the JS SDK? Or is there anything broken currently in the JS SDK?

Yes, plus, any dev who uses Mongo and explain will have to modify their code no matter if they use JS or or REST directly. I think the Flutter SDK might use explain also in which it will probably break. The only SDK that won’t break is the Swift SDK because I never conformed to the bug

**Comment by mtrezza** [comment]
Before we weight breaking all SDKs vs. making the Swift SDK conform to the bug - would it be possible at all to make the Swift SDK conform?

**Comment by cbaker6** [comment]
> Before we weigh breaking all SDKs vs. making the Swift SDK conform to the "bug" - would it be possible at all to make the Swift SDK conform?

Doing this would break the Postgres implementation in which I setup correctly on the server side. In its current form (before this PR), only Mongo or Postgres explain can work on strongly typed SDKs, not both as the Mongo explain was originally coded incorrectly

**Comment by cbaker6** [comment]
My point is, the Mongo storage adaptor is disobeying the correct sending of results.

when developing the Swift SDK I never conformed to bugs as long as I knew it was a bug originally. Of course, I could have introduced bugs which should be fixed over time.

**Comment by mtrezza** [comment]
Yes, I understand your point, it makes sense to improve these things. I am trying to understand the implication of such a change. If it breaks only developer code, then this is just another breaking change. But if it breaks SDKs, we may need to prepare these SDKs before making the change in the server. It would be interesting to know which SDKs exactly this would break.

**Comment by cbaker6** [comment]
> Yes, I understand your point, it makes sense to improve these things. I am trying to understand the implication of such a change. If it breaks only developer code, then this is just another breaking change. But if it breaks SDKs, we may need to prepare these SDKs before making the change in the server. It would be interesting to know which SDKs exactly this would break.

From memory, the JS, Swift (non-breaking), and Flutter SDKs have explain. I don’t think the others have it, but I’m on mobile, so we will have to confirm

**Comment by cbaker6** [comment]
For the JS SDK, this will not break the SDK itself, but breaks it's testcases. In addition, if devs were using find explain they will need to fix their code to use an array instead of an object.

**Comment by mtrezza** [comment]
Then I think we can go ahead with a PR and highlight the breaking change.

**Comment by mtrezza** [comment]
I classified this as a bug with severity S4 (small / trivial):
- It does not impact any existing functionality, but can be considered a "conceptual flaw" because it does not allow easily implement `explain` in SDKs with strongly typed languages.
- The workaround is to use the REST API.

**Comment by cbaker6** [comment]
Another workaround is to write an `afterFind` trigger in Cloud Code to take the object explain returns and place it in array. This should allow `explain` to work work with strongly typed SDKs like the Swift SDK.
